types/json/binary.go

```diff
@@ -392,7 +392,12 @@ func ParseBinaryFromString(s string) (bj BinaryJSON, err error) {
 		err = ErrInvalidJSONText.GenWithStackByArgs("The document is empty")
 		return
 	}
-	if err = bj.UnmarshalJSON(hack.Slice(s)); err != nil {
+	data := hack.Slice(s)
+	if !json.Valid(data) {
```

Comment:
seems we need to modify the valid function to return the first invalid position in the json value to make the error message the same as mysql.

Reply:
`json.Unmarshal` will return the first invalid position, can we use it?

Reply:
Yeah, but it seems a little expensive. Can we use a more light weighted way?

Reply:
shall we write a json validCheck  function by ourselves?

Reply:
I don't think that is trivial work to validate json by ourselves. If we really need to do this, it can be in another PR, what do u think? @zz-jason @XuHuaiyu

Reply:
```go
 18 // Valid reports whether data is a valid JSON encoding.
 19 func Valid(data []byte) bool {
 20     return checkValid(data, &scanner{}) == nil
 21 }
 22
 23 // checkValid verifies that data is valid JSON-encoded data.
 24 // scan is passed in for use by checkValid to avoid an allocation.
 25 func checkValid(data []byte, scan *scanner) error {
 26     scan.reset()
 27     for _, c := range data {
 28         scan.bytes++
 29         if scan.step(scan, c) == scanError {
 30             return scan.err
 31         }
 32     }
 33     if scan.eof() == scanError {
 34         return scan.err
 35     }
 36     return nil
 37 }
```

The above is the implementation of `json.Valid()`, I think we can directly return `ErrInvalidJSONText` in `checkValid`, and the position info is stored in `scan.bytes`.

Reply:
this is golang json library code, should we copy it to tidb and modify?

Reply:
Oops, I didn't noticed that. Then I think we can keep it simple, no need to modify the golang json library code to make the error message be compatible with mysql.